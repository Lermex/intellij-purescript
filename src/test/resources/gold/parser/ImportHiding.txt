Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ModuleName)
          PSASTWrapperElement(Qualified)
            PSASTWrapperElement(ModuleName)
              PSASTWrapperElement(TypeConstructor)
                PsiElement(proper name)('Control')
              PsiElement(.)('.')
            PSASTWrapperElement(ModuleName)
              PSASTWrapperElement(TypeConstructor)
                PsiElement(proper name)('Monad')
              PsiElement(.)('.')
            PSASTWrapperElement(ModuleName)
              PSASTWrapperElement(TypeConstructor)
                PsiElement(proper name)('Eff')
              PsiElement(.)('.')
            PsiElement(proper name)('Console')
        PsiElement(whitespace)('\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ModuleName)
          PSASTWrapperElement(Qualified)
            PsiElement(proper name)('Prelude')
        PsiElement(whitespace)(' ')
        PsiElement(hiding)('hiding')
        PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('show')
        PsiElement(,)(',')
        PsiElement(whitespace)(' -- a value\n  ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PsiElement(class)('class')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(ClassName)
            PSASTWrapperElement(Qualified)
              PsiElement(proper name)('Show')
        PsiElement(,)(',')
        PsiElement(whitespace)(' -- a type class\n  ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PsiElement(proper name)('Unit')
          PsiElement(()('(')
          PsiElement(..)('..')
          PsiElement())(')')
          PsiElement(whitespace)('  -- a constructor\n  ')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('show')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(NumericLiteral)
              PsiElement(float)('1.0')
              PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(TypeClassDeclaration)
        PsiElement(class)('class')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(Qualified)
            PsiElement(proper name)('Show')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(Implies)
          <empty list>
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(TypeDeclaration)
          PSASTWrapperElement(TypeAnnotationName)
            PsiElement(identifier)('noshow')
            PsiElement(whitespace)(' ')
          PsiElement(::)('::')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('a')
                PsiElement(whitespace)(' ')
            PsiElement(->)('->')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Type)
              PSASTWrapperElement(TypeAtom)
                PSASTWrapperElement(GenericIdentifier)
                  PsiElement(identifier)('a')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('Unit')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(proper name)('X')
          PsiElement(whitespace)(' ')
        PsiElement(|)('|')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(proper name)('Y')
          PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PsiElement(do)('do')
            PsiElement(whitespace)('\n  ')
            PSASTWrapperElement(DoNotationValue)
              PSASTWrapperElement(Value)
                PSASTWrapperElement(PrefixValue)
                  PSASTWrapperElement(Var)
                    PSASTWrapperElement(Qualified)
                      PsiElement(identifier)('logShow')
                      PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Var)
                    PSASTWrapperElement(Qualified)
                      PsiElement(identifier)('show')
                      PsiElement(whitespace)('\n  ')
            PSASTWrapperElement(DoNotationValue)
              PSASTWrapperElement(Value)
                PSASTWrapperElement(PrefixValue)
                  PSASTWrapperElement(Var)
                    PSASTWrapperElement(Qualified)
                      PsiElement(identifier)('log')
                      PsiElement(whitespace)(' ')
                  PSASTWrapperElement(StringLiteral)
                    PsiElement(string)('"Done"')