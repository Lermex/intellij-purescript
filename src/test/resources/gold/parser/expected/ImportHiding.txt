Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Console')
        PsiElement(whitespace)('\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Prelude')
        PsiElement(whitespace)(' ')
        PsiElement(hiding)('hiding')
        PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('show')
        PsiElement(,)(',')
        PsiElement(whitespace)(' -- a value\n  ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PsiElement(class)('class')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(ClassName)
            PSASTWrapperElement(Qualified)
              PsiElement(proper name)('Show')
        PsiElement(,)(',')
        PsiElement(whitespace)(' -- a type class\n  ')
        PSASTWrapperElement(PositionedDeclarationRef)
          PsiElement(proper name)('Unit')
          PsiElement(()('(')
          PsiElement(..)('..')
          PsiElement())(')')
          PsiElement(whitespace)('  -- a constructor\n  ')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('show')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(NumericLiteral)
              PsiElement(float)('1.0')
              PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeClassDeclaration)
        PsiElement(class)('class')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ClassName)
          PsiElement(proper name)('Show')
          PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(TypeDeclaration)
          PsiElement(identifier)('noshow')
          PsiElement(whitespace)(' ')
          PsiElement(::)('::')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeVar)
                PsiElement(identifier)('a')
                PsiElement(whitespace)(' ')
            PsiElement(->)('->')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Type)
              PSASTWrapperElement(TypeAtom)
                PSASTWrapperElement(TypeVar)
                  PsiElement(identifier)('a')
                  PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Unit')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('X')
        PsiElement(whitespace)(' ')
        PsiElement(|)('|')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Y')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PsiElement(do)('do')
            PsiElement(whitespace)('\n  ')
            PSASTWrapperElement(DoNotationValue)
              PSASTWrapperElement(Value)
                PSASTWrapperElement(PrefixValue)
                  PSASTWrapperElement(Var)
                    PSASTWrapperElement(Qualified)
                      PsiElement(identifier)('logShow')
                      PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Var)
                    PSASTWrapperElement(Qualified)
                      PsiElement(identifier)('show')
                      PsiElement(whitespace)('\n  ')
            PSASTWrapperElement(DoNotationValue)
              PSASTWrapperElement(Value)
                PSASTWrapperElement(PrefixValue)
                  PSASTWrapperElement(Var)
                    PSASTWrapperElement(Qualified)
                      PsiElement(identifier)('log')
                      PsiElement(whitespace)(' ')
                  PSASTWrapperElement(StringLiteral)
                    PsiElement(string)('"Done"')